### ISSUE TITLE
Change data type used by uv_gettimeofday()?

### ISSUE THREAD

**Issue by cjihrig** [libuv/libuv#2243]
See discussion around https://github.com/nodejs/node/pull/27029#discussion_r270737060 for context.

Since `uv_gettimeofday()` hasn't gone out in a release, we still have a chance to get this right. Should we create a new data type for `uv_gettimeofday()` that uses `int64_t`?

**Comment by refack** [comment]
IIUC the use of `uv_timeval_t` as the return value of `uv_gettimeofday` was a default decision based on the [POSIX API](https://pubs.opengroup.org/onlinepubs/009696799/functions/gettimeofday.html)
https://github.com/libuv/libuv/blob/99440bb6734bc7edfea108f6eb6914982e9f8cf7/include/uv.h#L1097-L1100
And the POSIX `timeval` was represented as two 32bit values because 32bit was the common architecture, and 64bit arithmetic wasn't part of C.

As pointed out in https://github.com/nodejs/node/pull/27029#discussion_r270737060 when trying to calculate the total value in milliseconds, the intuitive expression 
```c
ts.tv_sec * 1000 + ts.tv_usec / 1000
```
can overflow on 32bit machines (where [`long` is usually only 32 bit](https://en.cppreference.com/w/c/language/arithmetic_types) wide)
(For example current value of seconds since epoch is ~1554129670, which has 31 significant bits, so multiplying by 1000 (~2^10) requires ~ 41 bits)

tl;dr since a 64bit int (preferably unsigned) can safely hold even nanoseconds since epoch (~51 bits), it seems like it's the safest and most convenient type to return.
